pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -428,8 +428,7 @@ private BrokerRequest getRealtimeBrokerRequest(BrokerRequest hybridBrokerRequest
   private void attachTimeBoundary(String rawTableName, BrokerRequest brokerRequest, boolean isOfflineRequest) {
     TimeBoundaryService.TimeBoundaryInfo timeBoundaryInfo =
         _timeBoundaryService.getTimeBoundaryInfoFor(TableNameBuilder.OFFLINE.tableNameWithType(rawTableName));
-    if (timeBoundaryInfo == null || timeBoundaryInfo.getTimeColumn() == null
-        || timeBoundaryInfo.getTimeValue() == null) {
+    if (timeBoundaryInfo == null) {
```

Comment:
if we skipped setting the time boundary as in line 110 of HelixExternalviewBasedTimeBoundaryService.java, then is it still guaranteed that timeBoundaryInfo.getTimeValue() is non-null?

Reply:
If we skip setting time boundary, then timeBoundaryInfo will be null. It is guaranteed that time column and time value are non-null in timeBoundaryInfo.